### PR TITLE
Resolve #83, #84, #173

### LINKED ISSUE #83: set_missing_count treat USUBJID as a factor
### Description

set_missing_count() currently counts the NA's actually present in the data.
I'd like to display the missing count as the number of subjects who don't have an assessment at the current visit. 
This could be mocked up in the input data by joining a mock dataset of subjects, PARAMs, and AVISIT values so that the NA's insert themselves, but can Tplyr do this automatically?

### Example of Table (Feature Request Only)

For example below, that would be Missing = 1 for TRT B for each PARAM and AVISIT.

``` r
library(Tplyr)
library(tibble)
library(dplyr)

adsl <-
  tribble(~ USUBJID, ~ SAFFL, ~TRT01AN, ~TRT01A,
          "101-001", "Y",     1,        "TRT A",
          "101-002", "Y",     2,        "TRT B",
          "101-003", "Y",     1,        "TRT A",
          "101-004", "Y",     2,        "TRT B")

adpe <- tribble(~USUBJID,  ~AVISITN, ~AVISIT,     ~PARAMN, ~PARAM,   ~TRT01AN, ~TRT01A,    ~AVALC,   ~SAFFL,
                "101-001",  -10,      "Screening",  1,      "Head",    1,     "TRT A",  "Normal", "Y",
                "101-001",  -10,      "Screening",  2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                "101-001",  -1,       "Day -1",     2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                "101-001",   5,       "Day 5",      2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                
                "101-002",  -10,      "Screening",  1,      "Head",    2,     "TRT B",  "Normal", "Y",
                "101-002",  -10,      "Screening",  2,      "Lungs",   2,     "TRT B",  "Normal", "Y",
                "101-002",  -1,       "Day -1",     2,      "Lungs",   2,     "TRT B",  "Normal", "Y",
                "101-002",   5,       "Day 5",      2,      "Lungs",   2,     "TRT B",  "Normal", "Y",
                
                "101-003",  -10,      "Screening",  1,      "Head",    1,     "TRT A",  "Normal", "Y",
                "101-003",  -10,      "Screening",  2,      "Lungs",   1,     "TRT A",  "Abnormal", "Y",
                "101-003",  -1,       "Day -1",     2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                "101-003",   5,       "Day 5",      2,      "Lungs",   1,     "TRT A",  "Abnormal", "Y",
)

adpe$AVALC <- factor(adpe$AVALC, levels = c("Normal", "Abnormal"))

t <- tplyr_table(adpe, TRT01A, where = SAFFL == 'Y') %>%
  set_pop_data(adsl) %>%
  set_pop_treat_var(TRT01A) %>%
  set_pop_where(SAFFL =="Y") %>%
  add_layer(
    group_count(AVALC, by = vars(PARAM, AVISIT)) %>%
     set_distinct_by(USUBJID) %>%
     set_denoms_by(TRT01A) %>%
     set_missing_count(f_str("xxx", n), denom_ignore=FALSE, Missing = NA)
  ) 


dat<- build(t) %>%
  arrange(ord_layer_1, ord_layer_2, ord_layer_3)

```

### LINKED ISSUE #84: by groups which are actually in the data
### Description

Desired records could be filtered during post-processing, but is there an option to create only those by groups, which are actually available in the dataset?

### Steps to Reproduce (Bug Report Only)

``` r
library(Tplyr)
library(tibble)
library(dplyr)

adsl <-
  tribble(~ USUBJID, ~ SAFFL, ~TRT01AN, ~TRT01A,
          "101-001", "Y",     1,        "TRT A",
          "101-002", "Y",     2,        "TRT B",
          "101-003", "Y",     1,        "TRT A",
          "101-004", "Y",     2,        "TRT B")

adpe <- tribble(~USUBJID,  ~AVISITN, ~AVISIT,     ~PARAMN, ~PARAM,   ~TRT01AN, ~TRT01A,    ~AVALC,   ~SAFFL,
                "101-001",  -10,      "Screening",  1,      "Head",    1,     "TRT A",  "Normal", "Y",
                "101-001",  -10,      "Screening",  2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                "101-001",  -1,       "Day -1",     2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                "101-001",   5,       "Day 5",      2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                
                "101-002",  -10,      "Screening",  1,      "Head",    2,     "TRT B",  "Normal", "Y",
                "101-002",  -10,      "Screening",  2,      "Lungs",   2,     "TRT B",  "Normal", "Y",
                "101-002",  -1,       "Day -1",     2,      "Lungs",   2,     "TRT B",  "Normal", "Y",
                "101-002",   5,       "Day 5",      2,      "Lungs",   2,     "TRT B",  "Normal", "Y",
                
                "101-003",  -10,      "Screening",  1,      "Head",    1,     "TRT A",  "Normal", "Y",
                "101-003",  -10,      "Screening",  2,      "Lungs",   1,     "TRT A",  "Abnormal", "Y",
                "101-003",  -1,       "Day -1",     2,      "Lungs",   1,     "TRT A",  "Normal", "Y",
                "101-003",   5,       "Day 5",      2,      "Lungs",   1,     "TRT A",  "Abnormal", "Y",
)

adpe$AVALC <- factor(adpe$AVALC, levels = c("Normal", "Abnormal"))

t <- tplyr_table(adpe, TRT01A, where = SAFFL == 'Y') %>%
  set_pop_data(adsl) %>%
  set_pop_treat_var(TRT01A) %>%
  set_pop_where(SAFFL =="Y") %>%
  add_layer(
    group_count(AVALC, by = vars(PARAM, AVISIT)) %>%
     set_distinct_by(USUBJID) %>%
     set_denoms_by(TRT01A) 
  ) 

dat<- build(t) %>%
  arrange(ord_layer_1, ord_layer_2, ord_layer_3)

```

**Expected behavior:** [What you expected to happen]

For PARAM = 'Head' records created only for 'Screening' but not for 'Day -1' and 'Day 5'.

**Actual behavior:** [What actually happened]

Cartesian product of visits and params

### PR DESCRIPTION
**Pull request by mstackhouse**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

This PR adds the new function `set_limit_data_by()` to address #84. Along the way, a pre-existing bug (#173) was found and this PR resolves that issue as well and adds tests to catch it in the future. 

The last push also includes updates for #83. Note there are a few warnings coming out of `test-count.R`, but those should be resolved by #167 which removes the chunk of code from which is warnings are emanating.

### CODE REVIEW

File: vignettes/denom.Rmd [hunk 1]
```diff
@@ -233,6 +233,32 @@ t %>%
...
+        set_missing_subjects_row_label("Missing Subjects")
+    ) %>%
+    build()
+
+  tail(missing_subs) %>% 
+    select(-starts_with('ord')) %>% 
+    kable()
+```
+
+In the example above, we produce a nested count layer. Using the function `add_missing_subjects_row()`. This triggers the addition of the new result row for which the missing subjects are calculated. The row label applied for this can be configured using `set_missing_subjects_row_label()`, and the row label itself will default for Missing. Depending on your sorting needs, a `sort_value` can be applied to whatever numeric value you provide. Lastly, you can provide an `f_str()` to format the missing subjects row separately from the rest of the layer, but whatever format is applied to the layer will apply otherwise.
```
Comment (asbates) [inline vignettes/denom.Rmd:258]: ... nested count layer. The function `add_missing_subjects_row()` triggers the addition ...

... default for Missing. -> ... default to 'Missing.'

**Review by asbates — CHANGES_REQUESTED** [review]
Just a couple comments on some working in a vignette.

Other than that it looks good to me.

File: vignettes/denom.Rmd (hunk 1 above)
Comment (mstackhouse) [reply vignettes/denom.Rmd]: Ok that's updated.

**Review by asbates — APPROVED** [review]

File: R/count.R [hunk 2]
```diff
@@ -101,6 +108,7 @@ process_summaries.count_layer <- function(x, ...) {
 
     process_count_denoms(x)
 
+    outer <- FALSE
```
[the remark this answers is not in the document]
Comment (mstackhouse) [reply R/count.R]: This must be leftover from iterations of development - because removing it doesn't break any of my tests so should be good.

**Review by elimillera — APPROVED** [review]

**Pull request merged**